core/src/rpc.rs

```diff
@@ -953,7 +953,20 @@ impl JsonRpcRequestProcessor {
                             .load(Ordering::SeqCst)
                 {
                     let result = self.blockstore.get_complete_block(slot, true);
-                    return Ok(result.ok().map(|confirmed_block| {
+                    return Ok(result.ok().map(|mut confirmed_block| {
+                        if confirmed_block.block_time.is_none()
+                            || confirmed_block.block_height.is_none()
+                        {
+                            let r_bank_forks = self.bank_forks.read().unwrap();
+                            if let Some(bank) = r_bank_forks.get(slot) {
+                                if confirmed_block.block_time.is_none() {
+                                    confirmed_block.block_time = Some(bank.clock().unix_timestamp);
+                                }
+                                if confirmed_block.block_height.is_none() {
+                                    confirmed_block.block_height = Some(bank.block_height());
+                                }
+                            }
```

Comment:
@carllin , we actually weren't doing the BankForks fallback in `get_block()`, even though we were in `get_transaction()` and `get_block_time()`. Added now!

Reply:
Got it, so this should be fine as long as the statuses are written to before the `BankForks` new root purges these slots, ~16 seconds

Reply:
Exactly